### ISSUE TITLE
Enum types not parsed correctly in duckdb sql

### ISSUE THREAD

**Issue by joouha** [tobymao/sqlglot#3043]
DuckDB supports enum data type definitions at table creation, but sqlgot fails to parse this.

```python
>>> import sqlglot
... sqlglot.transpile("CREATE TABLE color ( name ENUM('RED','GREEN','BLUE'));", read="duckdb", write="sqlite")
```

```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".a/lib/python3.11/site-packages/sqlglot/__init__.py", line 164, in transpile
    for expression in parse(sql, read, error_level=error_level)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/__init__.py", line 87, in parse
    return Dialect.get_or_raise(read or dialect).parse(sql, **opts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/dialects/dialect.py", line 469, in parse
    return self.parser(**opts).parse(self.tokenize(sql), sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 1086, in parse
    return self._parse(
           ^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 1152, in _parse
    expressions.append(parse_method(self))
                       ^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 1354, in _parse_statement
    return self.STATEMENT_PARSERS[self._prev.token_type](self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 616, in <lambda>
    TokenType.CREATE: lambda self: self._parse_create(),
                                   ^^^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 1474, in _parse_create
    this = self._parse_schema(this=table_parts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 4275, in _parse_schema
    self._match_r_paren()
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 5891, in _match_r_paren
    self.raise_error("Expecting )")
  File "./lib/python3.11/site-packages/sqlglot/parser.py", line 1196, in raise_error
    raise error
sqlglot.errors.ParseError: Expecting ). Line 1, Col: 30.
  CREATE TABLE color ( name ENUM('RED','GREEN','BLUE'));
```

Defining ENUM types at table creation appears to work fine in duckdb v0.10.0:

```duckdb
D CREATE TABLE color ( name ENUM('RED', 'GREEN', 'BLUE'));
D SELECT column_name, data_type FROM duckdb_columns() where table_name = 'color';
┌─────────────┬──────────────────────────────┐
│ column_name │          data_type           │
│   varchar   │           varchar            │
├─────────────┼──────────────────────────────┤
│ name        │ ENUM('RED', 'GREEN', 'BLUE') │
└─────────────┴──────────────────────────────┘
```

**Comment by joouha** [comment]
Thanks!

I was just using sqlite as an example - I hadn't though about sqlite not supporting enums. 

The issue I was having concerned using [ibis](https://github.com/ibis-project/ibis) with duckdb databases with enum types, so hopefully that's fixed by #3044.

**Comment by georgesittas** [comment]
I see, thanks for clarifying - that should be fixed indeed.
